### PR TITLE
DEPS: Drop Python 3.4 support

### LINKED ISSUE #15251: DEPS: drop python 3.4

### PR DESCRIPTION
**Pull request by gfyoung**
Title is self-explanatory.

Closes #15251.

### CODE REVIEW

**Comment by jreback** [comment]
so we have a couple of circle ci jobs which test 3.4 move these to 3.6 i think.

**Comment by gfyoung** [comment]
@jreback : Good catch.  Copied and pasted the `requirements-3.6*` files to `requirements-3.6_SLOW*` files and also removed 3.4 from the `tox.ini` file (git diff is not very helpful, hence why I'm explaining my changes here).

**Comment by gfyoung** [comment]
@jreback : Weird locale failure on Circle <a href="https://circleci.com/gh/pandas-dev/pandas/1875#tests/containers/2">here</a>.

**Comment by gfyoung** [comment]
@jreback : Everything is green here, but I presume you are waiting to merge this until AFTER `0.20.2` is released.

**Comment by jreback** [comment]
@gfyoung no, we can merge things for 0.21.0 now. @jorisvandenbossche had an objection. I actually have no problem with this. yes its not *hard* to support. But 3.4 is not very well supported by conda anymore, no point in us holding out the torch.

**Comment by gfyoung** [comment]
@jreback : So is this good to go then?

File: ci/requirements-3.6.run [hunk 1]
```diff
@@ -14,7 +14,7 @@ html5lib
 jinja2
 sqlalchemy
 pymysql
-feather-format
+# feather-format (not available on defaults ATM)
```
Comment (jreback) [inline ci/requirements-3.6.run]: no, the 3.6 run specially uses conda-forge (this is the only run that does this)

Comment (gfyoung) [inline ci/requirements-3.6.run]: So just change the comment?

Comment (jreback) [inline ci/requirements-3.6.run]: revert the change
you shouldn't be changing this file 
it's for the travis build

Comment (gfyoung) [inline ci/requirements-3.6.run]: Circle broke when i left the file as is.  Couldn't install feather.

Comment (jreback) [inline ci/requirements-3.6.run]: u need a separate file (and a different job tag)
also ok to just have  3 circle  jobs

Comment (gfyoung) [inline ci/requirements-3.6.run]: Ah, okay.  That's works.  Will do then.

**Comment by gfyoung** [comment]
@jreback : Made the requested changes.  PTAL.

File: circle.yml [hunk 2]
```diff
@@ -23,9 +23,9 @@ dependencies:
        0)
          sudo apt-get install language-pack-it && ./ci/install_circle.sh JOB="2.7_COMPAT" LOCALE_OVERRIDE="it_IT.UTF-8" ;;
        1)
-         sudo apt-get install language-pack-zh-hans && ./ci/install_circle.sh JOB="3.4_SLOW" LOCALE_OVERRIDE="zh_CN.UTF-8" ;;
+         sudo apt-get install language-pack-zh-hans && ./ci/install_circle.sh JOB="3.6_SLOW" LOCALE_OVERRIDE="zh_CN.UTF-8" ;;
```
Comment (jreback) [inline circle.yml]: actuall let's name these

3.6_LOCALE_SLOW
3.6_LOCALE

I think it more consistent

Comment (gfyoung) [inline circle.yml]: Sounds good.  Done.

**Comment by gfyoung** [comment]
@jreback : Made the requested changes, and everything is still green.  PTAL.

File: ci/requirements-3.6_SLOW.build [hunk 3]
```diff
@@ -0,0 +1,6 @@
+python=3.6*
```
Comment (jreback) [inline ci/requirements-3.6_SLOW.build]: seems you added a 3.6_SLOW, pls remove.

Comment (gfyoung) [inline ci/requirements-3.6_SLOW.build]: Oops.  Done.

**Comment by gfyoung** [comment]
@jreback : Made the requested changes, and everything is still green. PTAL.

**Comment by jreback** [comment]
thanks!
